runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -347,22 +366,26 @@ def up(self):
             )
             task.set_resources(
                 sky.Resources(
-                    cloud=cloud_provider,
+                    cloud=cloud_provider,  # TODO: confirm if passing instance type in old way (without --) works when provider is k8s
+
                     instance_type=self.instance_type
                     if self.instance_type
                     and ":" not in self.instance_type
-                    and "CPU" not in self.instance_type
```

Comment:
+1, I think this will break passing `instance_type="CPU:2". What we want is to just check if `--` is in instance type (or self.instance_type), in which case we just pass it as is, and if not, use the existing behavior. We should probably just do this (compute the `instance_type` `accelerators` and `cpus`) above instead of inline here.